hldb-iOSTests/hldbTests.swift

```diff
@@ -554,4 +554,196 @@ class HLDBTests: XCTestCase {
+        "c" : "c5",
+        "d" : 3])
+      let row6 = HLDB.Table.Row(fields: ["id" : "row6",
+        "a"  : "a6",
+        "b" : 6,
+        "c" : "c6",
+        "d" : 3])
+      table.insert([row1, row2, row3, row4, row5, row6]).onSuccess { result in
+        
+        switch result {
+        case .Success:
+          break
+        case .Error(let code, let message):
+          XCTAssert(false, "Insert query returned error \(code) \(message)")
+        case .Items(let items):
+          XCTAssert(false, "Insert query returned items rather than success")
+        }
+        
+        // query to see if we can find those items
+        table.select().onSuccess { result in
+          switch result {
+          case .Success:
+            XCTAssert(false, "Insert query returned success rather than items")
+          case .Error(let code, let message):
+            XCTAssert(false, "Insert query returned error \(code) \(message)")
+          case .Items(let items):
+            
+            if items.count != 6 {
+              XCTAssert(false, "Expected six items, got \(items.count)")
+            }
+          }
+        }
+        
+        table.select(whereStr: "c='c3'").onSuccess { result in
+          switch result {
+          case .Success:
+            XCTAssert(false, "Insert query returned success rather than items")
+          case .Error(let code, let message):
+            XCTAssert(false, "Insert query returned error \(code) \(message)")
+          case .Items(let items):
+            
+            if items.count != 1 {
+              XCTAssert(false, "Expected one item!")
+            }
+          }
+        }
+        
+        table.select(whereStr: "d=3").onSuccess { result in
+          switch result {
+          case .Success:
+            XCTAssert(false, "Insert query returned success rather than items")
+          case .Error(let code, let message):
+            XCTAssert(false, "Insert query returned error \(code) \(message)")
+          case .Items(let items):
+            
+            if items.count != 2 {
+              XCTAssert(false, "Expected two items, got \(items.count)")
+            }
+          }
+        }
+        
+        table.select(whereStr: "d=2 AND c='c3'").onSuccess { result in
+          switch result {
+          case .Success:
+            XCTAssert(false, "Insert query returned success rather than items")
+          case .Error(let code, let message):
+            XCTAssert(false, "Insert query returned error \(code) \(message)")
+          case .Items(let items):
+            
+            if items.count != 1 {
+              XCTAssert(false, "Expected one item, got \(items.count)")
+            }
+          }
+        }
+        
+        //prove column b is unique
+        let row7 = HLDB.Table.Row(fields: ["id" : "row7",
+          "a"  : "a1",
+          "b" : 1,
+          "c" : "c1",
+          "d" : 1])
+        table.insert([row7]).onSuccess { result in
+          switch result {
```

Comment:
shouldn' we do a select after this to make sure that the unique aspect really worked?